### DISCUSSION TITLE
Add database metadata

### DISCUSSION THREAD

**Discussion by tobiasBora** [dexie/Dexie.js#1934] [General]
In my model, databases informally representing a file, this way I can easily add/import/export/remove files by simply doing the operation on the database. I'd like now to associate to each database a value, like the date of the creation, the name of the file, the version of the format it contains… I could create a table `fileInfo` with `id` equal to the name of the property… but this seems dirty, for instance this way I do not get typescript typing etc. Is there a better solution, that would ideally support typescript typing?

--EDIT--
Actually, I'm thinking that to get typing, it might be simpler to simply create a row with all parameters, and query the first row always… but curious to know if it is the cleanest way, notably this requires to open the database, which might not be very efficient?

**Comment by dfahlander** [comment]
I think using dexie that way is more cumbersome and also less performant than having one single database and let every file be represented by a row in one table of the database. You might want to list the databases or index them by some property.

**Reply by tobiasBora** [reply]
Yes, my main issue with this is that file properties will not be exported with the database this way (unless I duplicate this in the database, but I'm not a big fan of this as it breaks DRY). Or maybe I can move from one presentation to another only during export/import… might be a solution if I find this too inefficient in practice.

**Reply by ecofi** [reply]
Adding meta data would be very helpful. Is this already possible?

For example, adding meta data can be useful if the database should only persist for a specific time and should be be removed after x days. This could improve performance. Instead of putting large data all in one db. At the moment it seems only possible to store meta data outside of the db, i.e. localStorage, another db etc  (what is all not ideal when data outside is lost). Or to store this inside with a specific table.

Hence, I think it would be helpful if metadata about the db can be stored.
